feat(MFLP-80): Detaching the Store module and applying FeignClient

dereck-jun opened the pull request:
## 📌 PR 요약

- 가게 도메인 모듈 분리 및 FeignClient 적용

## 🔗 관련 이슈

- MFLP-80

## 🛠️ 작업 내역

- feign client와 dto를 모아두는 global-feign 모듈 추가
- 가게 도메인 모듈 분리

## 📸 스크린샷 (선택)

작업 결과를 스크린샷으로 첨부해주세요.

| 기능  | 스크린샷               |
|-----|--------------------|
| PNG | ![스크린샷](사진을 넣어주세요) |

## ⚠️ 주의 사항 (선택)

- Auth 모듈이 실행이 안돼서 동작 테스트를 진행하지 못했습니다.
- feign 정의는 global-feign에 정의하고, 사용할 땐 필요로 하는 모듈에서 정의된 feign을 상속받은 뒤 `@FeignClient`를 붙여서 속성을 넣어주시면 됩니다.
- global-feign에는 동작할 API 정의 및 DTO만 넣어주시고, 추가적인 설정이 필요할 경우 global-config 모듈에 적절하게 추가해주시면 됩니다.

## ✅ 체크리스트

PR 작성자가 확인해야 할 항목입니다:

- [ ] 코드가 정상적으로 동작하는지 확인했습니다.
- [x] 코드 리뷰어를 등록했습니다.
- [x] Labels를 등록했습니다.

dereck-jun committed:
feat(MFLP-80): Detaching the Store module and applying FeignClient

dereck-jun committed:
chore(MFLP-80): Adding missing dependencies to the Products module

KimKiHong-1111 reviewed (changes requested):
현재 생성, 다건 조회, 수정, 삭제는 작동확인.
단건 조회는 수정필요합니다

KimKiHong-1111 commented on buyhood-store/src/main/java/api/buyhood/store/service/StoreService.java:
```diff
@@ -0,0 +1,199 @@
+		}
+
+		if (StringUtils.hasText(address)) {
+			if (getStore.getName().equalsIgnoreCase(address)) {
```
getName이 아니고 getAddress 일거 같습니다

KimKiHong-1111 reviewed (changes requested):
고생많으셨습니다

dereck-jun committed:
fix(MFLP-80): Add `@RequestParam` to getListStoreCategoryResByIds

dereck-jun committed:
fix(MFLP-80): Fixing store category name lookup within a store domain

- Solving the auth module unable to run issue build.gradle in the user module
- Fix `getStoreCategoryResByNameOrElseThrow()`

dereck-jun committed:
refactor(MFLP-80): Modify the `getUserResOrElseThrow()` API endpoint

dereck-jun committed:
fix(MFLP-80): Modify store Fixes duplicate checks for store name instead of address when checking my address

KimKiHong-1111 committed:
Merge branch 'develop' into feat/MFLP-80
